plugins/modules/ec2_asg.py

```diff
@@ -1523,6 +1544,40 @@ def replace(connection):
     return changed, asg_properties
 
 
+def detach(connection):
+    group_name = module.params.get('name')
+    detach_instances = module.params.get('detach_instances')
+    as_group = describe_autoscaling_groups(connection, group_name)[0]
+    decrement_desired_capacity = module.params.get('decrement_desired_capacity')
+    min_size = module.params.get('min_size')
+    props = get_properties(as_group)
+    instances = props['instances']
+
+    # check if provided instance exists in asg, create list of instances to detach which exist in asg
+    instances_to_detach = []
+    for instance_id in detach_instances:
+        if instance_id in instances:
+            instances_to_detach.append(instance_id)
+
+    # check if setting decrement_desired_capacity will make desired_capacity smaller
+    # than the currently set minimum size in ASG configuration
+    if decrement_desired_capacity:
+        decremented_desired_capacity = len(instances) - len(instances_to_detach)
+        if min_size and min_size > decremented_desired_capacity:
+            module.fail_json(
+                msg="Detaching instance(s) with 'decrement_desired_capacity' flag set reduces number of instances below min_size,\
+                        please update AutoScalingGroup Sizes properly.")
```

Comment:
It would be nice to include here what the min_size is and how much we would have reduced desired capacity by, so the user knows how much they would need to reduce the size by.